src/obstacle_avoidance/src/MyNode.cpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Created By: Gareth Ellis
+ * Created On: July 16th, 2016
+ * Description: An example node that subscribes to a topic publishing strings,
+ *              and re-publishes everything it receives to another topic with
+ *              a "!" at the end
+ */
+
+#include <MyNode.h>
+
+sensor_msgs::LaserScan AvoidObstacleClass::lidar_data;
+
+int sizeOfLidarArray(sensor_msgs::LaserScan lidar);
+
+AvoidObstacleClass::AvoidObstacleClass() {
+  // initialise velocity
+  vel_msg.linear.x = LINEAR_VEL;
+  vel_msg.linear.y = 0;
+  vel_msg.linear.z = 0;
+  vel_msg.angular.x = 0;
+  vel_msg.angular.y = 0;
+  vel_msg.angular.z = 0;
+}
+
+AvoidObstacleClass::AvoidObstacleClass(int argc, char **argv,
+                                       std::string node_name) {
+  // Setup NodeHandles
+  ros::init(argc, argv, node_name);
+  nh = new ros::NodeHandle;
+
+  // Setup Subscriber/Publisher
+  twist_publisher = nh->advertise<geometry_msgs::Twist>("/cmd_vel", 10);
+  laser_scan_subscriber =
+      nh->subscribe("/scan", 10, &AvoidObstacleClass::lidarCallback, this);
+
+  // initialise velocity
+  vel_msg.linear.x = LINEAR_VEL;
+  vel_msg.linear.y = 0;
+  vel_msg.linear.z = 0;
+  vel_msg.angular.x = 0;
+  vel_msg.angular.y = 0;
+  vel_msg.angular.z = 0;
+  std::cout << "completed construction \n";
+  // std::cout << "lidar data " << lidar_data.angle_max << std::endl;
+  // std::cout << "lidar size " << sizeOfLidarArray(lidar_data) << std::endl;
+  // std::cout << lidar_data << std::endl;
```

Comment:
Print statements are OK for debugging purposes, but should be removed before finalising and pushing code. (even commented out ones)